File: common/src/main/java/com/ssblur/scriptor/word/descriptor/focus/inventory/CasterInventoryDescriptor.java
package com.ssblur.scriptor.word.descriptor.focus.inventory;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.helpers.targetable.ContainerTargetable;
import com.ssblur.scriptor.helpers.targetable.EntityTargetable;
import com.ssblur.scriptor.helpers.targetable.InventoryEntityTargetable;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.word.descriptor.focus.FocusDescriptor;
import net.minecraft.world.Container;

public class CasterInventoryDescriptor  extends Descriptor implements FocusDescriptor {
  @Override
  public Cost cost() {
    return new Cost(0, COSTTYPE.ADDITIVE);
  }

  @Override
  public Targetable modifyFocus(Targetable targetable) {
    if(targetable instanceof ContainerTargetable)
      return targetable;
    if(targetable instanceof EntityTargetable entityTargetable)
      return new InventoryEntityTargetable(entityTargetable.getTargetEntity(), 0);
    if(targetable.getLevel().getBlockEntity(targetable.getTargetBlockPos()) instanceof Container)
      return new ContainerTargetable(targetable.getLevel(), targetable.getTargetBlockPos(), 0);
    return targetable;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/focus/inventory/CasterInventoryDescriptor.kt
package com.ssblur.scriptor.word.descriptor.focus.inventory

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.ContainerTargetable
import com.ssblur.scriptor.helpers.targetable.EntityTargetable
import com.ssblur.scriptor.helpers.targetable.InventoryEntityTargetable
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.descriptor.focus.FocusDescriptor
import net.minecraft.world.Container

class CasterInventoryDescriptor : Descriptor(), FocusDescriptor {
    override fun cost(): Cost {
        return Cost(0.0, COSTTYPE.ADDITIVE)
    }

    override fun modifyFocus(targetable: Targetable): Targetable {
        if (targetable is ContainerTargetable) return targetable
        if (targetable is EntityTargetable) return InventoryEntityTargetable(targetable.targetEntity, 0)
        if (targetable.level.getBlockEntity(targetable.targetBlockPos) is Container)
            return ContainerTargetable(targetable.level, targetable.targetBlockPos, 0)
        return targetable
    }
}
